### ISSUE TITLE
Initramfs networking is broken in new builds

### ISSUE THREAD

**Issue by ajeddeloh** [coreos/fedora-coreos-tracker#313]
Current builds of Fedora CoreOS from testing-devel do not have working networking in the initramfs. Using an Ignition config that needs networking causes boot to fail. Running `dhclient` from the emergency shell also fails:
```
  624.115401] dhclient[515]: can't create /var/lib/dhclient/dhclient.leases: No such file or directory
[  624.133032] dhclient[516]: execve (/usr/sbin/dhclient-script, ...): No such file or directory
[  624.144703] dhclient[515]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 5 (xid=0xb2e51e76)
[  624.154166] dhclient[515]: send_packet: Network is down
[  624.159941] dhclient[515]: dhclient.c:2872: Failed to send 300 byte long packet over eth0 interface.
[  624.168788] dhclient[515]: receive_packet failed on eth0: Network is down
[  629.523621] dhclient[515]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 7 (xid=0xb2e51e76)
[  629.527405] dhclient[515]: send_packet: Network is down
[  629.528986] dhclient[515]: dhclient.c:2872: Failed to send 300 byte long packet over eth0 interface.
```

I hit this when testing live PXE and trying to fetch an Ignition config over tftp, but it fails when `cosa run` as well when trying to fetch a config to merge.

```
$ rpm-ostree status
* ostree://fedora:fedora/x86_64/coreos/testing-devel
                   Version: 30.20191104.dev.1 (2019-11-04T21:40:33Z)
                    Commit: 661e61f91eaa164829db4a8984ab6f7faf82da8cc8efac20c34b8c84c7c70be7
              GPGSignature: (unsigned)
```

**Comment by jlebon** [comment]
Hmm I initially thought this was related to bodhi-updates failing in the pipeline. There are a bunch of SSH connection failures, though `coreos.ignition.resource.remote` did pass.

Once we fix this, we should also figure out how this got into `testing-devel`. We might need to add a kola test. Or are you running with overrides/`cosa fetch --update-lockfile`?

**Comment by ajeddeloh** [comment]
I have an override for a custom ignition, but it's just a patch for testing on IBM cloud, shouldn't affect anything. I am running with `cosa fetch --update-lockfile`

**Comment by jlebon** [comment]
> I am running with `cosa fetch --update-lockfile`

Ahh OK, so it's likely the culprit is in `git diff`. Mind pasting it here?

(FWIW, I'd reword the first sentence as something like:

> Fedora CoreOS built from the latest updates do not have working networking in the initramfs.

I think it's important to distinguish between `testing-devel` being broken vs `testing-devel` + modifications/updates; the former is worse since it means it's a blocker for doing releases and signals a gap in testing or elsewhere in the process.)

**Comment by ajeddeloh** [comment]
I assume you mean the lockfile diff: https://gist.github.com/ajeddeloh/2f384e70bba49c3fc45172e3cc0361c8

I haven;t done a completely fresh build since the 30/31 switch so it's not super useful.

**Comment by ansemjo** [comment]
If I may chime in, I think I am also missing networking in the current PXE images somehow?

I'm attemting to try out FCOS by downloading PXE Installer `kernel` and `initramfs` (`30.20191014.0 testing`) locally to `$PWD` and creating a minimal `treeinfo`. Then I'm launching a virtual machine with `virt-install`:

```
virt-install --name fcos --memory 2048 --vcpus 2 --accelerate --autostart \
  --graphics none --disk size=20,bus=virtio --os-variant virtio26 --location $PWD \
  --extra-args "console=ttyS0 coreos.inst=yes coreos.inst.install_dev=vda coreos.inst.image_url=http://172.26.63.122/fcos/fcos.raw.xz coreos.inst.platform_id=qemu" \
  --qemu-commandline="-fw_cfg name=opt/com.coreos/config,file=$PWD/ignition.json"
```

It boots up fine but then fails to fetch the image:

    [    3.101327] coreos-installer[496]: failed fetching image headers from http://172.26.63.122/fcos/fcos.raw.xz: 7

Entering the emergency console it's clear why that is:

    :/# ip a show eth0
    2: eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
        link/ether 52:54:00:0d:46:f5 brd ff:ff:ff:ff:ff:ff

Is there another kernel commandline flag to explicitly enable `dhclient` on start or configure networking exclusively for the installer?

**Comment by jlebon** [comment]
OK, I'm going to assume this is the same issue that's blocking the pipeline right now. If it's not, I'll split out into a new bug. Trying an unlocked build locally to debug.

**Comment by jlebon** [comment]
OK, the pipeline issue turned out to be something else: https://github.com/coreos/fedora-coreos-config/pull/223.

I didn't encounter this in an unlocked build I did today. Can you try again on the latest?

> I haven;t done a completely fresh build since the 30/31 switch so it's not super useful.

OK, if it's still an issue, an up to date diff against `testing-devel` would help (though it's going to be noisy right now until we get the latest updates, which shouldn't be too long now).

**Comment by ajeddeloh** [comment]
Just built a fresh image, looks like it's fixed now. Closing.

I did discover an Ignition bug though: Ignition doesn't retry on tftp urls, but that belongs in the Ignition tracker and I'll file it there. The result is that specifying the config with `ignition.config.url=tftp://...` can cause failures to boot when Ignition is running as the network comes up.
